sdk/azidentity/azure_cli_credential.go

```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 )
 
+const azureCliCredentialTroubleshootMessage string = "\nTo troubleshoot, visit https://aka.ms/azsdk/go/identity/azclicredential/troubleshoot"
```

Comment:
In JavaScript we linked to each individual credential 🤔 I believe it was intentional. Let me ping @KarishmaGhiya and @scottaddie

Reply:
@chlowell If we go this route, will the fragment be the same as the fragment on the troubleshooting guide page? or do we have to redefine the fragments in the aka.ms section just like we have the links? I think if it's the former, then every time we change the content on the Troubleshooting guide, we'll have to modify the code too with the updated fragments in the error description (if there is). Maybe this was the reason we had separate links...

cc: @g2vinay I know we created different links for each credential so that we could guide the user to the exact sub-section on the troubleshooting guide. Is there any other reason?

Reply:
Do aka.ms links carry the hashtag over?

Reply:
My preference would be a single aka.ms link to rule them all (aka.ms/azsdk/go/identity/troubleshoot). We'd then do exactly what Charles has proposed. For example, https://aka.ms/azsdk/go/identity/troubleshoot#azclicredential.

Reply:
Ok I can do that! — @KarishmaGhiya should we then update on JS? (and Java, @g2vinay ?)

Reply:
So we did a quick verification by making a common aka.ms link for TS guide page and appending the url fragment in there - it does seem to work. 
https://aka.ms/azsdk/js/identity/troubleshoot#troubleshoot-azure-powershell-authentication-issues

But i have a few concerns - 
1. It makes the complete link very long. The links we have for each credential are short and concise and follow a standard structure.
2. There's a dependency on the content of the page. If the content and fragment changes - we need to update this in every error message in the code that the user sees v/s if we have the separate links, we'll just update the fragments in one place which is the aka.ms section without touching the code.

Reply:
> 1. It makes the complete link very long

You could shorten the anchor in the guide. For example, "Troubleshoot Azure PowerShell Authentication Issues" could be 
 simply "Azure PowerShell". Then the fragment link would be `https://aka.ms/azsdk/js/identity/troubleshoot#azure-powershell`, which is shorter than what you have now.

> 2. There's a dependency on the content of the page. If the content and fragment changes - we need to update this in every error message in the code

If you define a constant string for each link, there's only one place in the code to update. As a maintainer of code and aka.ms links, I believe keeping N strings in sync with guide anchors would be easier than maintaining N aka.ms links.

Reply:
Once again, Charles has read my mind. 😄 

What if you introduce an anchor tag for each section heading? That allows you to easily shorten. Using Azure PowerShell as an example:

```
<a name="az-pwsh"></a>
```

Your link for this credential type then becomes `https://aka.ms/azsdk/js/identity/troubleshoot#az-pwsh`. Whatever we decide to do here should be applied to the other languages for consistency.

Reply:
In about a week I’m expecting to come back to this. I got stuck with other things in JS.

The last time we spoke over teams with the Identity SDK crew, we discussed that using a single link with a hashtag won’t work if in the future the troubleshooting guideline is moved to the MS Docs website.

I don’t remember if we had a counter-argument to that. Should I keep the many aka.ms links instead of moving to the single aka.ms link?

Reply:
I'll counter argue that having a single aka link doesn't preclude a seamless transition to the docs site. We could add matching anchors in the new document.

Reply:
I don’t have a preference here. Let’s get an agreement, cc: @g2vinay , @scottaddie